dipy/reconst/tests/test_ivim.py

```diff
@@ -0,0 +1,191 @@
+""" Testing IVIM
+
+"""
+import numpy as np
+from numpy.testing import (assert_array_equal, assert_array_almost_equal)
+
+from dipy.reconst.ivim import ivim_function, IvimModel
+from dipy.core.gradients import gradient_table, generate_bvecs
+from dipy.sims.voxel import multi_tensor
+from dipy.core.sphere import disperse_charges, HemiSphere
+
+
+def test_fit_minimize():
+    """
+    Test the implementation of the fitting with minimize
+    """
+    bvals = np.array([0., 10., 20., 30., 40., 60., 80., 100.,
+                      120., 140., 160., 180., 200., 220., 240.,
+                      260., 280., 300., 350., 400., 500., 600.,
+                      700., 800., 900., 1000.])
+    N = len(bvals)
+    bvecs = generate_bvecs(N)
+    gtab = gradient_table(bvals, bvecs.T)
+
+    S0, f, D_star, D = 1.0, 0.132, 0.00885, 0.000921
+
+    mevals = np.array(([D_star, D_star, D_star], [D, D, D]))
+    # This gives an isotropic signal
+
+    signal = multi_tensor(gtab, mevals, snr=None, S0=S0, fractions=[
+                          f * 100, 100 * (1 - f)])
+    data = np.array([signal[0], ])
+
+    ivim_model = IvimModel(gtab)
+    ivim_fit = ivim_model.fit(data, routine="minimize")
+
+    est_signal = np.array([ivim_function(ivim_fit.model_params[0], bvals), ])
+
+    assert_array_equal(est_signal.shape, data.shape)
+    assert_array_almost_equal(est_signal, data)
+    assert_array_almost_equal(ivim_fit.model_params[0], [S0, f, D_star, D])
+
+
+def test_fit_leastsq():
+    """
+    Test the implementation of the fitting with leastsq
+    """
+    bvals = np.array([0., 10., 20., 30., 40., 60., 80., 100.,
+                      120., 140., 160., 180., 200., 220., 240.,
+                      260., 280., 300., 350., 400., 500., 600.,
+                      700., 800., 900., 1000.])
+    N = len(bvals)
+    bvecs = generate_bvecs(N)
+    gtab = gradient_table(bvals, bvecs.T)
+
+    S0, f, D_star, D = 1.0, 0.132, 0.00885, 0.000921
+
+    mevals = np.array(([D_star, D_star, D_star], [D, D, D]))
+    # This gives an isotropic signal
+
+    signal = multi_tensor(gtab, mevals, snr=None, S0=S0, fractions=[
+                          f * 100, 100 * (1 - f)])
```

Comment:
I thought about that but I read that unit tests should usually be orthogonal and independent and in most of the other tests there are no global variables.